src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -1181,6 +1212,47 @@ private void addNewFolder() {
 		}
 	}
 
+	private void renameFolder() {
+		renameFolder.setEnabled(true);
+		IElement selected = list.getSelectedValue();
+		String newName = VOptionPane
+				.showInputDialog(mcreator, L10N.t("workspace.elements.folders.rename.message"), L10N.t("workspace.elements.folders.rename.title"), null, new OptionPaneValidatior() {
+					@Override public ValidationResult validate(JComponent component) {
+						String newName = ((JTextField) component).getText();
+						if (!newName.matches("[A-Za-z0-9._ -]+")) {
+							canRename = false;
+							return new Validator.ValidationResult(ValidationResultType.ERROR, L10N.t("workspace.elements.folders.add.error_letters"));
+						} else {
+							canRename = true;
+							return Validator.ValidationResult.PASSED;
+						}
+					}
+				});
+		if (canRename == true) {
+			String finalNewName = newName;
+			list.getSelectedValuesList().forEach(re -> {
+				FolderElement folder = (FolderElement) re;
+				for (ModElement modElement : mcreator.getWorkspace().getModElements()) {

```

Comment:
To me it seems this code does not handle level 2 folder handling, did you test what happens when you rename parent folder of nested folder structure?

Reply:
what do you mean?

Reply:
Did you test what happens when you rename parent folder of nested folder structure? Try it out an see if it works and if not you will need to improve this part

Reply:
if you mean, that I rename a Folder inside a folder inside a folder?, well it works

Reply:
If you rename the first folder, what does happen with folder inside this folder inside this folder?

Reply:
they stay inside inside the renamed folder, like you saw in the image i sent up there

Reply:
Ok I will do some tests